File: Tests/Categories/TestAVFoundation.m
#import "AVAudioSession+AnyPromise.h"
@import AVFoundation;
@import XCTest;


@interface TestAVAudioSession: XCTestCase @end @implementation TestAVAudioSession

- (void)testM {
    id ex = [self expectationWithDescription:@""];

    [[AVAudioSession new] requestRecordPermission].then(^{
        [ex fulfill];
    });

    [self waitForExpectationsWithTimeout:1 handler:nil];
}

@end

File: Tests/Categories/TestAVFoundation.swift
import AVFoundation
import PromiseKit
import XCTest

class Test_AVAudioSession_Swift: XCTestCase {
    func test() {
        let ex = expectation(description: "")

        AVAudioSession().requestRecordPermission().then { _ in
            ex.fulfill()
        }

        waitForExpectations(timeout: 1, handler: nil)
    }
}
